### DISCUSSION TITLE
Does SYCL CUDA support compute-sanitizer?

### DISCUSSION THREAD

**Discussion by AllanZyne** [intel/llvm#14970] [Q&A] [answered]
https://developer.nvidia.com/compute-sanitizer
I guess compute-sanitizer does instrument directly on PTX.
Can we compile SPIRV image to PTX by AOT?

**Comment by al42and** [comment, marked as answer]
Do you mean SYCL code compiled with `-fsycl-targets=nvidia_gpu_sm_*`? Yes, that works quite well with NVIDIA's compute-sanitizer (I haven't tested if it picks up debug info, but it detects data races etc). But it compiles directly to PTX, without intermediate SPIR-V, if I understand correctly.
